### PR TITLE
Helm: Add service labels and annotations

### LINKED ISSUE #8627: Support adding annotations to all service's templates
I want to use "[Topology Aware Hints](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/)" feature of Kubernetes, but currently the helm chart templatefor `service` only support adding annotation for [gateway](https://github.com/grafana/loki/blob/9f8aa4b98a0048c658f49a6f9c3ad76e115efc84/production/helm/loki/templates/gateway/service-gateway.yaml#L12-L14) nodes.

Could you please add annotation support to all other targets like read/write/backend?

### PR DESCRIPTION
**Pull request by rubenvw-ngdata**
**What this PR does / why we need it**:
Support setting labels and annotations for all services in the Helm chart.
Move label configuration for read, write and gateway services to #component.service.labels to improve
consistency between services.

**Which issue(s) this PR fixes**:
Fixes #8627 

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] `CHANGELOG.md` updated
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

**Comment by rubenvw-ngdata** [comment]
Is this going to be reviewed soon? We are waiting for this functionality.

**Review by MasslessParticle — APPROVED** [review]
LGTM, Thanks for contributing!

**Comment by rubenvw-ngdata** [comment]
@MichelHollands Any update on this? Anything I should still do?

**Comment by MichelHollands** [comment]
> @MichelHollands Any update on this? Anything I should still do?

@rubenvw-ngdata Sorry, I'm busy with some other things. Can you check why the [CI run](https://github.com/grafana/loki/actions/runs/6074010143/job/16477025165?pr=10214) fails? If that is fixed it can be merged.

File: production/helm/loki/templates/backend/service-backend-headless.yaml [hunk 1]
```diff
@@ -8,7 +8,13 @@ metadata:
   namespace: {{ $.Release.Namespace }}
   labels:
     {{- include "loki.backendSelectorLabels" . | nindent 4 }}
+    {{- toYaml .Values.loki.serviceLabels | nindent 4}}
+    {{- toYaml .Values.singleBinary.service.labels | nindent 4 }}
```
Comment (trevorwhitney) [inline production/helm/loki/templates/backend/service-backend-headless.yaml:12]: why are we putting the single binary's service labels on the backend service? should this be backend instead?

File: production/helm/loki/templates/backend/service-backend-headless.yaml [hunk 2]
```diff
@@ -8,7 +8,13 @@ metadata:
...
+    variant: headless
     prometheus.io/service-monitor: "false"
+  annotations:
+    {{- toYaml .Values.loki.serviceAnnotations | nindent 4}}
+    {{- toYaml .Values.singleBinary.service.annotations | nindent 4 }}
```
Comment (trevorwhitney) [inline production/helm/loki/templates/backend/service-backend-headless.yaml:17]: same with the annotations?

**Review by trevorwhitney — CHANGES_REQUESTED** [review]
The templates are currently failing to render, at least with `ci/default-values.yaml`. Are you able to run a `helm template` locally using those values? I've tried and it fails, seems to be a problem with the format of the yaml templates as I keep getting the `error converting YAML to JSON` error.

File: production/helm/loki/templates/backend/service-backend.yaml [hunk 3]
```diff
@@ -8,9 +8,11 @@ metadata:
   namespace: {{ $.Release.Namespace }}
   labels:
     {{- include "loki.backendLabels" . | nindent 4 }}
-    {{- with .Values.backend.serviceLabels }}
-    {{- toYaml . | nindent 4 }}
-    {{- end }}
+    {{- toYaml .Values.loki.serviceLabels | nindent 4}}
```
Comment (trevorwhitney) [inline production/helm/loki/templates/backend/service-backend.yaml:11]: I think these may be failing to render when empty. Wrapping them in `with` statements should help

File: production/helm/loki/templates/backend/service-backend-headless.yaml (hunk 1 above)
Comment (rubenvw-ngdata) [reply production/helm/loki/templates/backend/service-backend-headless.yaml:12]: thx, fixing

File: production/helm/loki/templates/backend/service-backend-headless.yaml (hunk 2 above)
Comment (rubenvw-ngdata) [reply production/helm/loki/templates/backend/service-backend-headless.yaml:17]: fixing

**Comment by rubenvw-ngdata** [comment]
@trevorlauder I think the helm template issue is an issue that was recently introduced on main.
I'm getting errors now too, but I'm pretty certain that I have tested this and there were no issues.
Getting the same issue when I run it on the current main branch.

**Comment by MichelHollands** [comment]
@rubenvw-ngdata You are right: it was a problem on the main branch.

File: production/helm/loki/templates/backend/service-backend.yaml (hunk 3 above)
Comment (rubenvw-ngdata) [reply production/helm/loki/templates/backend/service-backend.yaml:11]: Can you give an example when you think this fails? I tested it and I don't see it causing issues. Without the `with` statement it's just a lot shorter.

Comment (rubenvw-ngdata) [reply production/helm/loki/templates/backend/service-backend.yaml]: anyhow, I have changed it to use with, doesn't really matter too much to me.

**Comment by rubenvw-ngdata** [comment]
@trevorwhitney @MichelHollands I think everything is ok now. Let me know if I should do anything else.

**Review by MichelHollands — APPROVED** [review]
LGTM

**Comment by MichelHollands** [comment]
> @trevorwhitney @MichelHollands I think everything is ok now. Let me know if I should do anything else.

@rubenvw-ngdata Thanks for your help and sorry for the hold-up. It is merged now.

**Pull request merged**